Automattic/jetpack
# My Jetpack: Avoid user error on connection status card when is not needed

Fixes #23054

#### Changes proposed in this Pull Request:
* Introduces new prop `requiresUserConnection` to connection status card
* Updates My Jetpack to use this property
* Updates tests to comply
* Introduces tests for when `requiresUserConnection` equals `false`.

#### Does this pull request change what data or activity we track or use?



no
#### Testing instructions:
* Admin 1 - None user connection, and some plugin requires a connection (Should see a red message about requires connection)
* Admin 1 - None user connection, and none plugin requires a connection (Should see a regular message without being red, asking to connect user)
* Admin 2 - None user connection, and some plugin requires a connection (Should see a red message about requires connection)
* Admin 2 - None user connection, and none plugin requires a connection (Should see a regular message without being red, asking to connect user)
* Admin 2 - Admin 1 user is connected, and itself is not, and some plugin requires a connection (Should not see connection message)
* Admin 2 - Admin 1 user is connected, and itself is not, and none plugin requires a connection (Should not see connection message)
